src/index/base.cpp

```diff
@@ -106,10 +104,9 @@ bool BaseIndex::Init()
         SetBestBlockIndex(locator_index);
     }
 
-    // Skip pruning check if indexes are not ready to sync (because reindex-chainstate has wiped the chain).
     const CBlockIndex* start_block = m_best_block_index.load();
     bool synced = start_block == active_chain.Tip();
-    if (!synced && g_indexes_ready_to_sync) {
```

Comment:
In commit "init: don't start indexes sync thread prematurely" (480f5aad060f4f60b5d0617cfb96276891ac692a)

It's not really clear why this change is safe. It seems like in case of reindexing maybe `active_chain.Tip()` could be null and the prune_violation error could happen below? Or possibly there's a reason that wouldn't happen that could be clarified here. Maybe `&& g_indexes_ready_to_sync` needs to be replaced with `&& !fReindexChainState`? Either way would at least be good to keep a code comment here about how the prune_violation check is supposed to work with reindexing.

Reply:
Hmm ok, this should have been part of the commits ordering changes.

The change is not safe in 480f5aad, it becomes safe at the last commit (in b9fe9ab). 

Basically, this last commit moves the pruning violation check from `Init()` to the end
of the 'loadblk' thread, which is where the reindex, block loading and chain activation
processes happen.

So we can run this verification step even when the reindex or reindex-chainstate flags
are enabled (which has being skipped so far).

Will add a check inside the intermediate commit so the history stay consistent and also
expand the last commit description to include this explanation. Thanks!

Reply:
Ah yeah. 
The other possibility, which adds a bit more overhead, could be to move the pruning violation check from `Init()` to `Start()` in the intermediate commit, then move it from `Start()` to `StartIndexes()` in the last one.

Will go with your suggestion. Thanks.